android 14 requiring more permissions again

targetSDK=34 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_MEDIA_PROJECTION] any of the permissions allOf=false [android.permission.CAPTURE_VIDEO_OUTPUT, android:project_media] 

```
FATAL EXCEPTION: main
Process: com.meshcentral.agent2, PID: 18580
java.lang.RuntimeException: Unable to start service com.meshcentral.agent.ScreenCaptureService@98247a with Intent { cmp=com.meshcentral.agent2/com.meshcentral.agent.ScreenCaptureService (has extras) }: java.lang.SecurityException: Starting FGS with type mediaProjection callerApp=ProcessRecord{5b4c7a 18580:com.meshcentral.agent2/u0a699} targetSDK=34 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_MEDIA_PROJECTION] any of the permissions allOf=false [android.permission.CAPTURE_VIDEO_OUTPUT, android:project_media] 
	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:5286)
	at android.app.ActivityThread.-$$Nest$mhandleServiceArgs(Unknown Source:0)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2531)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:230)
	at android.os.Looper.loop(Looper.java:319)
	at android.app.ActivityThread.main(ActivityThread.java:8913)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:608)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1103)
Caused by: java.lang.SecurityException: Starting FGS with type mediaProjection callerApp=ProcessRecord{5b4c7a 18580:com.meshcentral.agent2/u0a699} targetSDK=34 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_MEDIA_PROJECTION] any of the permissions allOf=false [android.permission.CAPTURE_VIDEO_OUTPUT, android:project_media] 
	at android.os.Parcel.createExceptionOrNull(Parcel.java:3069)
	at android.os.Parcel.createException(Parcel.java:3053)
	at android.os.Parcel.readException(Parcel.java:3036)
	at android.os.Parcel.readException(Parcel.java:2978)
	at android.app.IActivityManager$Stub$Proxy.setServiceForeground(IActivityManager.java:7214)
	at android.app.Service.startForeground(Service.java:775)
	at com.meshcentral.agent.ScreenCaptureService.onStartCommand(ScreenCaptureService.kt:354)
	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:5268)
	at android.app.ActivityThread.-$$Nest$mhandleServiceArgs(Unknown Source:0) 
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2531) 
	at android.os.Handler.dispatchMessage(Handler.java:106) 
	at android.os.Looper.loopOnce(Looper.java:230) 
	at android.os.Looper.loop(Looper.java:319) 
	at android.app.ActivityThread.main(ActivityThread.java:8913) 
	at java.lang.reflect.Method.invoke(Native Method) 
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:608) 
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1103) 
Caused by: android.os.RemoteException: Remote stack trace:
	at com.android.server.am.ActiveServices.validateForegroundServiceType(ActiveServices.java:2729)
	at com.android.server.am.ActiveServices.setServiceForegroundInnerLocked(ActiveServices.java:2440)
	at com.android.server.am.ActiveServices.setServiceForegroundLocked(ActiveServices.java:1785)
	at com.android.server.am.ActivityManagerService.setServiceForeground(ActivityManagerService.java:15860)
	at android.app.IActivityManager$Stub.onTransact$setServiceForeground$(IActivityManager.java:11873)
```

--- a/app/src/main/AndroidManifest.xml
+++ b/app/src/main/AndroidManifest.xml
@@ -12,6 +12,7 @@
     <uses-permission android:name="android.permission.READ_MEDIA_AUDIO" />
     <uses-permission android:name="android.permission.READ_MEDIA_VIDEO" />
     <uses-permission android:name="android.permission.FOREGROUND_SERVICE" />
+    <uses-permission android:name="android.permission.FOREGROUND_SERVICE_MEDIA_PROJECTION" />
     <uses-permission android:name="com.google.android.gms.permission.AD_ID" tools:node="remove"/>
 
     <application

--- a/app/src/main/java/com/meshcentral/agent/ScreenCaptureService.kt
+++ b/app/src/main/java/com/meshcentral/agent/ScreenCaptureService.kt
@@ -394,8 +394,6 @@ class ScreenCaptureService : Service() {
                     mOrientationChangeCallback!!.enable()
                 }
 
-                // Register media projection stop callback
-                mMediaProjection!!.registerCallback(this.MediaProjectionStopCallback(), mHandler)
                 g_ScreenCaptureService = this
                 updateTunnelDisplaySize()
                 sendAgentConsole("Started display sharing")
@@ -432,6 +430,8 @@ class ScreenCaptureService : Service() {
 
         // Start capture reader
         mImageReader = ImageReader.newInstance(mWidth, mHeight, PixelFormat.RGBA_8888, 2)
+        // Register media projection stop callback
+        mMediaProjection!!.registerCallback(this.MediaProjectionStopCallback(), mHandler)
         mVirtualDisplay = mMediaProjection!!.createVirtualDisplay(ScreenCaptureService.Companion.SCREENCAP_NAME, mWidth, mHeight,
                 mDensity, ScreenCaptureService.Companion.virtualDisplayFlags, mImageReader!!.surface, null, mHandler)
 

